pastas/model.py

```diff
@@ -611,8 +613,9 @@ def initialize(self, tmin=None, tmax=None, freq=None, warmup=None,
             self._get_time_offset(self.settings["freq"])
 
         # Set tmin and tmax
-        self.settings["tmin"] = self.get_tmin(tmin)
-        self.settings["tmax"] = self.get_tmax(tmax)
+        self.settings["tmin"] = self.get_tmin(
+            tmin).floor(self.settings["freq"])
+        self.settings["tmax"] = self.get_tmax(tmax).ceil(self.settings["freq"])
```

Comment:
This might be confusing because it can differ from the settings defined by the user. What do you think @raoulcollenteur ?

Reply:
Good point, perhaps the floor should then be moved into the `get_tmin()` method, and only applied when `tmin=None`.

And there is still a risk that when providing a tmin or tmax, the calculation will crash because the first observation lies before the first simulation timestep... Maybe we should allow extrapolation up to a timestep(=freq) away?